### PR TITLE
Add Steam Multiblocks and Potin

### PR DESCRIPTION
**Pull request by Irgendwer01**
**What:**
This PR ports the Steam Multiblocks and Potin from Gregicality (Tech22's PR) to GTCEU with a few modifications.

**Additional info:**
It's an bit messy but shouldn't be too messy to be unreadable.

**Possible compatibility issue:**
none I can aware of.

### CODE REVIEW

File: src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java
Comment (htmlcsjs) [inline src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java:64]: you could fix the bug mentioned here :wink:

File: src/main/java/gregtech/integration/jei/multiblock/infos/SteamOvenInfo.java
Comment (htmlcsjs) [inline src/main/java/gregtech/integration/jei/multiblock/infos/SteamOvenInfo.java:39]: I think these should allow any hatch, or at least ULV and LV

File: src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java
Comment (Irgendwer01) [reply src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java:64]: > you could fix the bug mentioned here 😉
I will do that in another PR since I don't know if it will break other things.

File: src/main/java/gregtech/integration/jei/multiblock/infos/SteamOvenInfo.java
Comment (Irgendwer01) [reply src/main/java/gregtech/integration/jei/multiblock/infos/SteamOvenInfo.java:39]: hmm, maybe

Comment (Irgendwer01) [reply src/main/java/gregtech/integration/jei/multiblock/infos/SteamOvenInfo.java:39]: From Discord:

> It was intended to be only steam

File: src/main/java/gregtech/common/metatileentities/steam/multiblockpart/MetaTileEntitySteamHatch.java
Comment (TechLord22) [inline src/main/java/gregtech/common/metatileentities/steam/multiblockpart/MetaTileEntitySteamHatch.java:58]: Please add `this.setPaintingColor(0xFFFFFF)` after initializeInventory(). It prevents tinting of this hatch and causing multiblocks to look slightly wrong if the feature is utilized.

File: src/main/java/gregtech/common/metatileentities/steam/multiblockpart/MetaTileEntitySteamItemBus.java
Comment (TechLord22) [inline src/main/java/gregtech/common/metatileentities/steam/multiblockpart/MetaTileEntitySteamItemBus.java:31]: Please add `this.setPaintingColor(0xFFFFFF)`. It prevents tinting of this hatch and causing multiblocks to look slightly wrong if the feature is utilized. Also, why is initializeInventory() not called here as well?

File: src/main/java/gregtech/common/ConfigHolder.java
Comment (TechLord22) [inline src/main/java/gregtech/common/ConfigHolder.java:15]: Please change this config name to "steelSteamMultiblocks". It's more clear on its purpose.

File: src/main/java/gregtech/common/ConfigHolder.java
Comment (TechLord22) [inline src/main/java/gregtech/common/ConfigHolder.java:19]: Please change this config name to "multiblockSteamToEU". It's more clear on its purpose.

File: src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java
Comment (TechLord22) [inline src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java:102]: You don't need this method. Just call `workable.update()` instead.

**Review by TechLord22 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java
Comment (htmlcsjs) [reply src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java:64]: Ok, good idea

File: src/main/java/gregtech/integration/jei/multiblock/infos/SteamOvenInfo.java
Comment (htmlcsjs) [reply src/main/java/gregtech/integration/jei/multiblock/infos/SteamOvenInfo.java:39]: Ok

File: src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java
Comment (Irgendwer01) [reply src/main/java/gregtech/api/capability/impl/SteamMultiblockRecipeLogic.java:65]: solved in Discord.

File: src/main/java/gregtech/api/unification/material/Materials.java
Comment (TechLord22) [reply src/main/java/gregtech/api/unification/material/Materials.java]: You forgot to change Potin's ID to 353.

**Review by TechLord22 — APPROVED** [review]

**Review by DStrand1 — APPROVED** [review]
Approved, but we should wait until #43 is merged to not cause extra merge conflicts in `Materials` for that PR since it is a much more complex change

**Pull request merged**
